test/e2e/kubevirt.go

```diff
@@ -1619,6 +1625,32 @@ write_files:
 				},
 			}
 
+			virtualMachineWithUDNAndStaticIPsAndMAC = resourceCommand{
+				description: "VirtualMachine with interface binding for UDN and statics IPs and MAC",
+				cmd: func() string {
+					GinkgoHelper()
+					if !isPreConfiguredUdnAddressesEnabled() {
+						Skip("ENABLE_PRE_CONF_UDN_ADDR not configured")
+					}
+					staticIPs, err := json.Marshal(map[string][]string{
+						"net1": filterCIDRs(fr.ClientSet, staticIPv4, staticIPv6),
+					})
+					Expect(err).ToNot(HaveOccurred())
+					annotations := map[string]string{
+						kubevirtAddressesAnnotation: string(staticIPs),
+					}
+					vm = fedoraWithTestToolingVM(nil /*labels*/, annotations, nil, /*nodeSelector*/
+						kubevirtv1.NetworkSource{
+							Pod: &kubevirtv1.PodNetwork{},
+						}, userDataWithIperfServer, networkDataDualStack)
+					vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].Bridge = nil
+					vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].Binding = &kubevirtv1.PluginBinding{Name: "l2bridge"}
+					vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress = staticMAC
```

Comment:
nit: maybe just setting this once using an `Interface` struct ? you'd then have a well named helper return that `Interface`, something like: `vm.Spec.Template.Spec.Domain.Devices.Interfaces[0] = primaryUDNIface(withMACAddr(staticMAC))`

you can ignore this comment.